### PR TITLE
WIP: mon: centralized config

### PR DESCRIPTION
**Pull request by liewegas**
The basics are kind of working!

- [x] use normal monclient auth for initial config
- [x] incorporate into global_init/common_init infrastructure
- [x] --no-mon-config argument
- [x] ceph config set WHO KEY VALUE
- [x] ceph config get WHO
- [x] ceph config get WHO KEY (includes default value)
- [x] ceph config dump
- [x] ceph config show WHO
- [x] ceph config show-with-defaults WHO
- [x] ceph config show WHO KEY
- [x] ceph config help
- [x] report mgr running config to mgr
- [x] annotate conf-only options (paxos_* maybe?)
- [x] make ConfigMonitor refresh more efficient than reparsing all config
- [x] show config option level in dump, get output
- [x] ceph config assimilate-conf -i old-ceph.conf -o new-ceph.conf
- [x] sort out ceph-conf behavior
- [x] 'option can be changed' in get and/or show (json?) output
- [x] docs
- [ ] clean up tell|daemon config set output (gross json that makes no sense)
- [x] tests

### CODE REVIEW

File: PendingReleaseNotes [hunk 1]
```diff
@@ -57,3 +57,10 @@
 * The RBD C API's rbd_discard method now enforces a maximum length of
   2GB to match the C++ API's Image::discard method. This restriction
   prevents overflow of the result code.
+
+13.0.1
+------
+
+* The format of the 'config diff' output via the admin socket has changed.  It
```
Comment (amitkumar50) [inline PendingReleaseNotes]: 2 spaces b/w fullstop and it.

File: PendingReleaseNotes [hunk 2]
```diff
@@ -57,3 +57,10 @@
...
+  now reflects the source of each config option (e.g., default, config file,
```
Comment (amitkumar50) [inline PendingReleaseNotes]: eg.,
. and , together

File: src/common/config.h [hunk 3]
```diff
@@ -95,12 +98,12 @@ struct md_config_t {
...
    */
-  std::map<std::string, Option::value_t> values;
+  std::map<std::string, map<int,Option::value_t>> values;
 
-   typedef enum {
-	OPT_INT, OPT_LONGLONG, OPT_STR, OPT_DOUBLE, OPT_FLOAT, OPT_BOOL,
-	OPT_ADDR, OPT_U32, OPT_U64, OPT_UUID
-   } opt_type_t;
+  typedef enum {
+    OPT_INT, OPT_LONGLONG, OPT_STR, OPT_DOUBLE, OPT_FLOAT, OPT_BOOL,
```
Comment (amitkumar50) [inline src/common/config.h]: NIT:tab

**Comment by liewegas** [comment]
jenkins render docs

**Comment by liewegas** [comment]
oops wrong branch, see https://github.com/ceph/ceph/pull/20172

**Pull request closed without merging**